lib/pure/httpclient.nim

```diff
@@ -970,7 +970,8 @@ proc newConnection(client: HttpClient | AsyncHttpClient,
       else: nativesockets.Port(connectionUrl.port.parseInt)
 
     when client is HttpClient:
-      client.socket = await net.dial(connectionUrl.hostname, port)
+      client.socket = newSocket()
```

Comment:
I think you might need to handle IPV4 and IPV6 here when creating `newSocket`. `dial` handles IPv4 and IPv6 automatically

Reply:
@ringabout I tried to find out how to do that but I got lost. I see that the async version is not explicitly using client.timeout either. Does it make sense to set timeout on operations which are using the socket (like listening/reading) down the line?